idea-plugin/src/main/kotlin/org/jetbrains/compose/intentions/utils/composableFinder/ComposableFunctionFinderImpl.kt

```diff
@@ -0,0 +1,41 @@
+package org.jetbrains.compose.intentions.utils.composableFinder
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.compose.intentions.utils.isPsiElementComposable.IsPsiElementComposable
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtPropertyDelegate
+import org.jetbrains.kotlin.psi.KtValueArgumentList
+import org.jetbrains.kotlin.psi.psiUtil.getChildOfType
+
+class ComposableFunctionFinderImpl : ComposableFunctionFinder, IsPsiElementComposable {
+
+    override fun isFunctionComposable(psiElement: PsiElement): Boolean {
+        return when (psiElement) {
+            is KtNameReferenceExpression -> psiElement.isComposable()
```

Comment:
What do you think about limiting the scope of intentions to just `KtNameReferenceExpression`?
I.e. dropping `KtCallExpression`, `KtProperty` and `KtValueArgumentList`.
I'm just not sure that the current implementation is intuitive and coherent.

For example:
```
Composable1(modifier = Modifier) {}
```
In the snippet above, the intentions are available at `Composable1` (which makes intuitive sense), at `(` and `)`, but not at `modifier = Modifier` and not at `{}`.

There are lots of types of elements in Kotlin PSI, I'm not sure that the current implementation will behave consistently. At the same time it complicates the behavior of the intention as well. For example, I can't clearly articulate, what `GetRootPsiElement` does, and why it is written in the way it is without taking into account this function. Narrowing the scope of intentions can make it easier to both use intentions and reason about code

Reply:
@AlexeyTsvetkov 
We can remove the KtCallExpression check. As per your example, it currently works on 'Composable' and '('. We would need 'KtValueArgumentList' for detecting on () like you said. {} is not included, of course.

KtProperty can be skipped, but you'll lose the ability to use it with properties. Currently, you can use it on 'val/var', 'property name', and the composable function like 'remember/animateAsFloat'.   If you wish to take that call, I can remove it.

So, what GetRootPsiElement does is, if you invoke it on a PsiElement, it gives you the parent PsiElement. 

For a composable function, it can be the KtCallExpression.
For a delegatedProperty, it will be KtProperty.

The root psi element is them wrapped with / deleted respectively. You can check out the GetRootPsiElementTest for more info